app/models/bet.rb

```diff
@@ -2,4 +2,41 @@ class Bet < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
   has_many :options, through: :user_bets, source: :bet_option
+
+  def resolve
+    losers.each do |debtor|
+      debt = amount_bet(debtor)
+      win_contributions.each do |contribution|
+        debtor.debts.create(creditor: contribution[:creditor],
```

Comment:
Another Comment on this method is What does it return. 

When designing methods you should try and think about them as atomic black boxes; code outside the method cannot see what is happening inside the method (even though they are not). Think:
- What information needs to go in
- What information needs to come out

We try and avoid side effects when writing methods. A side effect is when a method changes something in the system  with no indication externally that something has changed. In the case of this method. There is no indication outside of this method that debts are created.

Side effects make code hard to understand and hard to test. If I have a method:

```ruby
def add(num1, num2
  puts "#{num1 + num2}"
end

# and 

def add(num1, num2)
  num1 + num2
end
```

The second method is easier to test. I would write something like:
`expect(add(1,2)).to eq 3` 

But for the first one, the addition happens and the result is printed to the screen as a side effect. Much harder to test.

Think about that this method doing and what it should return in order to reduce side-effects happening within the blackbox of the method.